staging/src/k8s.io/apiserver/pkg/server/filters/hsts.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"net/http"
+	"strings"
+)
+
+// WithHsts is a simple HSTS implementation that wraps an http Handler.
+// If hstsHeaders is empty or nil, no HSTS support is installed.
+func WithHsts(handler http.Handler, hstsHeaders []string) http.Handler {
+	if len(hstsHeaders) == 0 {
+		return handler
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Chrome and Mozilla Firefox maintain an HSTS preload list
+		// issue : golang.org/issue/26162
+		// Set the Strict-Transport-Security header if it is not already set
+		if _, ok := w.Header()["Strict-Transport-Security"]; !ok {
+			w.Header().Set("Strict-Transport-Security", strings.Join(hstsHeaders, "; "))
```

Comment:
the value appears to be fixed.  Save us the allocations on the individual requests and compute the join ahead of time please.  Around line 29 I think.